incubator/mongodb/README.md

```diff
@@ -0,0 +1,192 @@
+| `ImagePullPolicy`     | Container pull policy     | `Always`                                               |
+| `Replicas`     | k8s petset replicas          | `3`                                                      |
+| `Component`    | k8s selector key                 | `mongodb`                                           |
+| `Cpu`          | container requested cpu          | `100m`                                                   |
+| `Memory`    |container requested memory                 | `512Mi`                                           |
+| `ClientPort`  | k8s service port                 | `2379`                                                   |
+| `PeerPorts`| Container listening port         | `27017`                                                   |
+| `Storage`| Persistent volume size         | `10Gi`                                                   |
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`.
+
+Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,
+
+```bash
+$ helm install --name my-release -f values.yaml mongodb-x.x.x.tgz
+```
+
+> **Tip**: You can use the default [values.yaml](values.yaml)
+
+Once you have all 3 nodes in running, you can run the "test.sh" script in this directory, which will insert a key into the primary and check the secondaries for output.
+
+# Deep dive
+
+## Cluster Health
+
+```
+$ for i in <0..n>; do kubectl exec <release-podname-$i> -- sh -c '/usr/bin/mongo --eval="printjson(db.serverStatus())"'; done
+```
+
+## Failover
+
+One can check the roles being played by each node by using the following:
+```console
+$ for i in <0..n>; do kubectl exec <release-podname-$i> -- sh -c '/usr/bin/mongo --eval="printjson(rs.isMaster())"'; done
+
+MongoDB shell version: 3.2.9
+connecting to: test
+{
+	"hosts" : [
+		"messy-hydra-mongodb-0.messy-hydra-mongodb.default.svc.cluster.local:27017",
+		"messy-hydra-mongodb-1.messy-hydra-mongodb.default.svc.cluster.local:27017",
+		"messy-hydra-mongodb-2.messy-hydra-mongodb.default.svc.cluster.local:27017"
+	],
+	"setName" : "rs0",
+	"setVersion" : 3,
+	"ismaster" : true,
+	"secondary" : false,
+	"primary" : "messy-hydra-mongodb-0.messy-hydra-mongodb.default.svc.cluster.local:27017",
+	"me" : "messy-hydra-mongodb-0.messy-hydra-mongodb.default.svc.cluster.local:27017",
+	"electionId" : ObjectId("7fffffff0000000000000001"),
+	"maxBsonObjectSize" : 16777216,
+	"maxMessageSizeBytes" : 48000000,
+	"maxWriteBatchSize" : 1000,
+	"localTime" : ISODate("2016-09-13T01:10:12.680Z"),
+	"maxWireVersion" : 4,
+	"minWireVersion" : 0,
+	"ok" : 1
+}
+
+
+```
+This lets us see which member is primary.
+
+Let us now test persistence and failover. First, we insert a key (in the below example, we assume pod 0 is the master):
+```console
+$ kubectl exec <$release-mongodb-0> -- /usr/bin/mongo --eval="printjson(db.test.insert({key1: 'value1'}))"
```

Comment:
Based on your example can we just use the name of the pod instead of `<$release-mongodb-0>`?  Or set it as a ENV variable, then they can set it to their own pod name?